qiling/loader/elf.py

```diff
@@ -615,4 +615,24 @@ def load_with_ld(self, stack_addr, load_address = -1, argv = [], env = {}):
         self.new_stack = new_stack
         self.load_address = load_address
         self.images.append(self.coverage_image(load_address, load_address + mem_end, self.path))
-        self.ql.os.function_hook = FunctionHook(self.ql, self.elf_phdr + mem_start, self.elf_phnum, self.elf_phent, load_address, load_address + mem_end)
\ No newline at end of file
+        self.ql.os.function_hook = FunctionHook(self.ql, self.elf_phdr + mem_start, self.elf_phnum, self.elf_phent, load_address, load_address + mem_end)
+
+        # map vsyscall section for some specific needs
+        if self.ql.archtype == QL_ARCH.X8664 and self.ql.ostype == QL_OS.LINUX:
+            _vsyscall_addr = int(self.ql.os.profile.get("OS64", "vsyscall_address"), 16)
+            _vsyscall_size = int(self.ql.os.profile.get("OS64", "vsyscall_size"), 16)
+
+            if not self.ql.mem.is_mapped(_vsyscall_addr, _vsyscall_size):
+                # initialize with \xcc then insert syscall entry
+                # each syscall should be 1KiB(0x400 bytes) away
+                self.ql.mem.map(_vsyscall_addr, _vsyscall_size, info="[vsyscall]")
+                self.ql.mem.write(_vsyscall_addr, _vsyscall_size * b'\xcc')
+
+                _vsyscall_entry_asm = [0x60c0c748, # gettimeofday
```

Comment:
i assume this is opcode and can we use ql.compile here ?

Reply:
its doable , but I found that the result of ql.compile need to be adjusted to fit the size, will make it more complicated

Reply:
try if possible. so code will be more readable. else stick to this is fine.

Reply:
i did try , doesnt look good to me. I recommend stick to this , and i will comment the asm after each one, is that acceptable ?

Reply:
just figured out how to use ql.compile correctly, it should be more readable now.